pkg/apis/core/types.go

```diff
@@ -2693,6 +2693,12 @@ type PodSpec struct {
 	// If not specified, the pod will not have a domainname at all.
 	// +optional
 	Subdomain string
+	// If set to true, the fully qualified domain name (FQDN) of a Pod is set as hostname of its containers.
+	// In Linux containers, this means setting the FQDN in the hostname field of the kernel (the nodename field of struct utsname).
+	// In Windows containers, this means setting the COMPUTERNAME to FQDN.
+	// If a pod does not have FQDN, this has no effect.
+	// +optional
+	HostnameFQDN bool
```

Comment:
Looking at existing bool fields, they start with a verb (such as Enable).

How about naming this field SetHostnameAsFQDN ?

Reply:
I am open to change it. During the KEP review, I was suggested to use `hostnameFQDN` or `fqdnHostname` ( see https://github.com/kubernetes/enhancements/pull/1792#discussion_r427535399). @thockin What do you think about changing the field as @tedyu suggest?

Reply:
moving conversation to https://github.com/kubernetes/kubernetes/pull/91699#discussion_r434222491 since we decide to close this pr